Feature-015(Desarrollar PetService)

SergioZF09 opened the pull request:
[Feature-015: Agregado los servicios CRUD de la entidad Pet](https://trello.com/c/wiHbPpYP/31-feature-015desarrollar-petservice)

También se agregaron las exceptions que agregó Matías Nehuen para usarlas en los servicios de la entidad Pet.

SergioZF09 committed:
Feature-015: Agregado los servicios CRUD de la entidad Pet

SergioZF09 committed:
Merge branch 'history-013' into feature-015

matiasnm reviewed (commented):
Sergio agregale @Transactional a los metodos del servicio, sobre todo los que escriben o modifican la base de datos. los que no pueden omitirse pero mejor si llevan @Transactional(readOnly = true). Esta anotacion aegura que todo se haga en una sola transaccion en la base de datos y que si hay algun problema se hace un roolback automatico

SergioZF09 committed:
Update PetServiceImpl.java

Se le agregaron los decoradores @Transactional a cada uno de los servicios, incluyendo el readOnly. Y tambien se agrego @AllArgsConstructor para usar los repository y el mapper en los tests.

SergioZF09 commented:
> Sergio agregale @transactional a los metodos del servicio, sobre todo los que escriben o modifican la base de datos. los que no pueden omitirse pero mejor si llevan @transactional(readOnly = true). Esta anotacion aegura que todo se haga en una sola transaccion en la base de datos y que si hay algun problema se hace un roolback automatico

Listo, ya los agregué, y también agregué @AllArgsConstructor para que no haya inconvenientes en los tests. Muchas gracias por comentármelo

matiasnm commented:
> > Sergio agregale @transactional a los metodos del servicio, sobre todo los que escriben o modifican la base de datos. los que no pueden omitirse pero mejor si llevan @transactional(readOnly = true). Esta anotacion aegura que todo se haga en una sola transaccion en la base de datos y que si hay algun problema se hace un roolback automatico
> 
> Listo, ya los agregué, y también agregué @AllArgsConstructor para que no haya inconvenientes en los tests. Muchas gracias por comentármelo

Joya Sergio. Comentario sobre AllArgsConstructor, sirve para no usar @Autowired y en vez inyectar las dependencias al service a traves del constructor. (Que lo crea lombok en automatico). Autowired tmb funciona por supuesto👍

Christian-Cachero commented on src/main/java/com/equipo5/backend/serviceImpl/PetServiceImpl.java:
```diff
@@ -40,8 +42,57 @@ public PetResponseDTO createPet(PetRequestDTO petRequestDTO) {
+
+        if (petRequestDTO.breed() != null) petNotModified.setBreed(petRequestDTO.breed());
+
+        if (petRequestDTO.ownerId() != null) petNotModified.setOwner(owner);
```
por que la mascota puede cambiar de dueño?

Christian-Cachero reviewed (changes requested):
por que la mascota puede cambiar de dueño? fijate en el método update.

Tiene realmente sentido eso?

SergioZF09 committed:
Update PetServiceImpl.java

Se eliminó en el update el cambiar el id del usuario

SergioZF09 commented:
> por que la mascota puede cambiar de dueño? fijate en el método update.
> 
> Tiene realmente sentido eso?

Cierto, no se puede cambiar, solo que en el frontend en el formulario para registrar tiene como campo el id del usuario (owner), por eso lo había incluído para editar xd. Por ahorita lo eliminé, gracias por comentármelo Christian

Christian-Cachero reviewed (approved):
ahora sí.
